amqp/Go/amqpConsumer.go

```diff
@@ -0,0 +1,215 @@
+var MQ_ENDPOINTS MQEndpoints
+
+//Parsing the command line arguments
+func init() {
+    flag.StringVar(&queueName, "q", "DEMO", "Queue")
+    flag.StringVar(&topicName, "t", "TOP", "Topic")
+    flag.Parse()
+    if flag.NFlag() > 2 {
+        valid = false
+    } else if flag.NFlag() == 1 {
+        flag.Visit(func(f *flag.Flag) {
+            if f.Name == "q" || f.Name == "t" {
+                valid = true
+            }
+            if f.Name == "q"{
+                destination = queueName
+                isQueue = true
+            }else{
+                destination = topicName
+            }
+        })
+    } else {
+        valid = false
+    }
+
+    if valid{
+        parseEnv()
+    }
+}
+
+func main() {
+    if !valid || queueName=="" || topicName==""{
+        fmt.Println("Invalid arguments")
+        return
+    }
+    fmt.Println("Application is Starting...")
+
+    //Defining the AMQP URI
+    uri:= "amqp://"+EnvSettings.Username+":"+EnvSettings.Password+"@"+EnvSettings.Host+":"+EnvSettings.Port
+
+    // Set up a new connection
+    conn, err := connect(uri)
+    if err != nil{
+        log.Fatal("Error in creating connection: ", err)
+    }
+
+    // Create a session
+    session, err := createSession(conn)
+    if err != nil{
+        log.Fatal("Error in creating a session: ", err)
+    }
+
+    //When the destination is a queue, capability must be set to "queue". If not, topic is selected by default.
+    ReceiverOptions := &amqp.ReceiverOptions{
+        SourceCapabilities: []string{"topic"},
+    }
+    if isQueue{
+        ReceiverOptions = &amqp.ReceiverOptions{
+            SourceCapabilities: []string{"queue"},
+        }
+    }
+
+    // Create a Receiver
+    Receiver, err := createReceiver(session, ReceiverOptions)
+    if err != nil{
+        log.Fatal("Error in creating a Receiver: ", err)
+    }
+
+    //Recieve the messages
+    err = receiveMessages(Receiver)
+    if err != nil{
+        fmt.Println("Error while receiving messages: ",err)
+    }
+    fmt.Println("End of Sample amqpConsumer.go")
+}
+
+func connect(uri string) (*amqp.Conn, error){
+    conn, err := amqp.Dial(context.TODO(), uri, nil)
+    if err != nil {
+        return nil, err
+    }
+    fmt.Println("Connected to host.")
+    return conn, nil
+}
+
+func createSession(conn *amqp.Conn) (*amqp.Session, error){
+    session, err := conn.NewSession(context.TODO(), nil)
+    if err != nil {
+        return nil, err
+    }
+    fmt.Println("Session created.")
+    return session, nil
+}
+
+func createReceiver(session *amqp.Session, receiverOptions *amqp.ReceiverOptions) (*amqp.Receiver, error){
+    receiver, err := session.NewReceiver(context.TODO(), destination, receiverOptions)
+    if err != nil {
+        return nil, err
+    }
+    fmt.Println("Receiver mapped to destination.")
+    fmt.Println("Destination :",receiverOptions.SourceCapabilities)
+    return receiver, nil
+}
+
+func receiveMessages(receiver *amqp.Receiver) error{
+    for{

```

Comment:
This looks like a tight get loop. Is it possible for the client to wait for a period of time for a message to appear?

Reply:
By default, if there is no activity, ie. no frames being exchanged between the server and client, the connection will be forcibly closed in 15 seconds. This is due to the timeout negotiated when the connection was established. This can be modified by providing ConnOptions (https://pkg.go.dev/github.com/Azure/go-amqp#ConnOptions).

So, in a pub/sub case too, the same timeout will be considered and the publisher should begin to send frames before it expires.